packages/nx/src/project-graph/utils/retrieve-workspace-files.ts

```diff
@@ -76,6 +77,29 @@ export function retrieveProjectConfigurations(
   );
 }
 
+export async function retrieveProjectConfigurationsWithPartialResult(
+  plugins: LoadedNxPlugin[],
+  workspaceRoot: string,
+  nxJson: NxJsonConfiguration
+): Promise<ConfigurationResult> {
+  let projConfigs: ConfigurationResult;
+  try {
+    projConfigs = await retrieveProjectConfigurations(
+      plugins,
+      workspaceRoot,
+      nxJson
+    );
+  } catch (e) {
+    // Errors are okay for this because we're only running 1 plugin
+    if (e instanceof ProjectConfigurationsError) {
+      projConfigs = e.partialProjectConfigurationsResult;
+    } else {
+      throw e;
+    }
+  }
+  return projConfigs;
+}
+
```

Comment:
Can we keep this in add-plugin, or a similar file next to it in devkit? We shouldn't be using this inside `nx`, so I don't want to put it here.